protocol/messenger_communities_sharding_test.go

```diff
@@ -0,0 +1,161 @@
+	community, chat := createCommunity(&s.Suite, s.owner)
+
+	advertiseCommunityToUserOldWay(&s.Suite, community, s.owner, s.alice)
+	joinCommunity(&s.Suite, community, s.owner, s.alice, &requests.RequestToJoinCommunity{CommunityID: community.ID()}, "")
+
+	// Members should be able to receive messages in a community with sharding enabled.
+	{
+		_, err := s.owner.SetCommunityShard(&requests.SetCommunityShard{
+			CommunityID: community.ID(),
+			Shard: &shard.Shard{
+				Cluster: shard.MainStatusShardCluster,
+				Index:   64,
+			},
+		})
+		s.Require().NoError(err)
+
+		_, err = WaitOnMessengerResponse(s.alice, func(mr *MessengerResponse) bool {
+			return len(mr.communities) > 0 && mr.Communities()[0].Shard() != nil && mr.Communities()[0].Shard().Index == 64
+		}, "shard info not updated")
+		s.Require().NoError(err)
+
+		message := buildTestMessage(*chat)
+		_, err = s.owner.SendChatMessage(context.Background(), message)
+		s.Require().NoError(err)
+
+		_, err = WaitOnMessengerResponse(s.alice, func(mr *MessengerResponse) bool {
+			return len(mr.messages) > 0 && mr.Messages()[0].ID == message.ID
+		}, "message not received")
+		s.Require().NoError(err)
+	}
+
+	// Members should be able to receive messages in a community where the sharding configuration has been edited.
+	{
+		_, err := s.owner.SetCommunityShard(&requests.SetCommunityShard{
+			CommunityID: community.ID(),
+			Shard: &shard.Shard{
+				Cluster: shard.MainStatusShardCluster,
+				Index:   128,
+			},
+		})
+		s.Require().NoError(err)
+
+		_, err = WaitOnMessengerResponse(s.alice, func(mr *MessengerResponse) bool {
+			return len(mr.communities) > 0 && mr.Communities()[0].Shard() != nil && mr.Communities()[0].Shard().Index == 128
+		}, "shard info not updated")
+		s.Require().NoError(err)
+
+		message := buildTestMessage(*chat)
+		_, err = s.owner.SendChatMessage(context.Background(), message)
+		s.Require().NoError(err)
+
+		_, err = WaitOnMessengerResponse(s.alice, func(mr *MessengerResponse) bool {
+			return len(mr.messages) > 0 && mr.Messages()[0].ID == message.ID
+		}, "message not received")
+		s.Require().NoError(err)
+	}
+
+	// Members should continue to receive messages in a community if sharding is disabled after it was previously enabled.
+	{
+		_, err := s.owner.SetCommunityShard(&requests.SetCommunityShard{
+			CommunityID: community.ID(),
+			Shard:       nil,
+		})
+		s.Require().NoError(err)
+
+		_, err = WaitOnMessengerResponse(s.alice, func(mr *MessengerResponse) bool {
+			return len(mr.communities) > 0 && mr.Communities()[0].Shard() == nil
+		}, "shard info not updated")
+		s.Require().NoError(err)
+
+		message := buildTestMessage(*chat)
+		_, err = s.owner.SendChatMessage(context.Background(), message)
+		s.Require().NoError(err)
+
+		_, err = WaitOnMessengerResponse(s.alice, func(mr *MessengerResponse) bool {
+			return len(mr.messages) > 0 && mr.Messages()[0].ID == message.ID
+		}, "message not received")
+		s.Require().NoError(err)
```

Comment:
This code is repeated 3 times with a few changes, could we extract it to a function with parameters?